ghost/members-api/lib/repositories/MemberRepository.js

```diff
@@ -281,7 +282,29 @@ module.exports = class MemberRepository {
             memberStatusData.status = 'comped';
         }
 
-        // Subscribe member to default newsletters
+        //checks for custom signUp forms
+        if (memberData.type === 'subscribe' && memberData.newsletters) {
+            const allValidNewsletters = await this._newslettersService.browse({limit: 'all'});
+            const allValidNewslettersIds = allValidNewsletters.map(newsletter => newsletter.id);
+
+            if (memberData.newsletters) {
```

Comment:
I think this conditional is redundant - We already check `memberData.newsletters` is `truthy` in the parent conditional